### PR TITLE
fix: deprecate `set_output` from GitHub workflow

### LINKED ISSUE #2067: `check-label` workflow: The `set-output` command is deprecated
### What happened?

The following warning is logged as an annotation in each run of the check-label workflow.

```
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information, go to https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

### Steps to reproduce

Check the logs of this action, like [this](https://github.com/jaegertracing/jaeger-ui/actions/runs/7315751908)

### Expected behavior

No warning should be logged. 

### Relevant log output

_No response_

### Screenshot

_No response_

### Additional context

_No response_

### Jaeger backend version

_No response_

### SDK

_No response_

### Pipeline

_No response_

### Stogage backend

_No response_

### Operating system

_No response_

### Deployment model

_No response_

### Deployment configs

_No response_

### PR DESCRIPTION
**Pull request by EshaanAgg**
## Which problem is this PR solving?
Fixes #2067 

## Description of the changes
Removes the `set-output` command from the workflow

## How was this change tested?
NA

## Checklist
- [X] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [X] I have signed all commits
- [X] I have added unit tests for the new functionality
- [X] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by yurishkuro** [comment]
it's failing
```
Run is_dependabot=$(echo EshaanAgg | grep -o 'dependabot')
  is_dependabot=$(echo EshaanAgg | grep -o 'dependabot')
  echo "is_dependabot=$is_dependabot" >> $GITHUB_OUTPUT
  shell: /usr/bin/bash -e {0}
```
could it be that grep exits with error code?

**Comment by yurishkuro** [comment]
yep
```shell
$ echo foo | grep bar

 12:14:55 🔴 
```

File: .github/workflows/ci-label-check.yml [hunk 1]
```diff
@@ -16,7 +16,9 @@ jobs:
     steps:
       - name: Check PR author
         id: check_author
-        run: echo "::set-output name=is_dependabot::$(echo ${{ github.event.pull_request.user.login }} | grep -o 'dependabot')"
+        run: |
+          is_dependabot=$(echo ${{ github.event.pull_request.user.login }} | grep -o 'dependabot')
```
Comment (yurishkuro) [inline .github/workflows/ci-label-check.yml:20]:
```suggestion
          is_dependabot=$(echo ${{ github.event.pull_request.user.login }} | grep -o 'dependabot' || true)
```

File: .github/workflows/ci-label-check.yml [hunk 2]
```diff
@@ -16,7 +16,9 @@ jobs:
...
+          echo "is_dependabot=$is_dependabot" >> $GITHUB_OUTPUT
 
       - name: Check PR label
         if: steps.check_author.outputs.is_dependabot != 'dependabot'
```
Comment (yurishkuro) [inline .github/workflows/ci-label-check.yml:24]: wonder why we couldn't just do this here:
```suggestion
        if: github.event.pull_request.user.login != 'dependabot'
```

Comment (EshaanAgg) [reply .github/workflows/ci-label-check.yml:24]: This seems much better IMO rather than `grep`ing manually.

**Comment by EshaanAgg** [comment]
@yurishkuro Sorry for so many extraneous commits! This is the indented functionality right?

**Review by yurishkuro — APPROVED** [review]
This is much cleaner! Thanks.

**Pull request merged**